src/shared/Paramable.ts

```diff
@@ -112,10 +112,9 @@ export interface ParamableInterface {
     /**
      * assignParameters() should copy the value property of each item in
      * params to its proper place to be used in the implementing object.
-     * (Typically, to top-level properties of the object.) The reason that
-     * implementations should not used the values directly from the params
-     * object is that they should only be used in a state when validate()
-     * is succeeding.
+     * (Typically, to top-level properties of the object.) Implementations
+     * should not use the values directly from the params object, because
+     * parameters should only be used when validate() succeeds.
```

Comment:
I agree that the previous wording of this comment was pretty bad. However, I would request that you keep the emphatic semantics that parameters should only be used in a situation where in that specific situation, validate() succeeds. The point here is that parameters should by most code be considered "volatile", since they are capriciously changed by the person manipulating the visualization. Hence, validate() succeeding at one point does not mean that forever thereafter it is OK to use the parameters.
I suspect you can find an equally nice and reasonably succinct rewording that maintains that nuance.

Reply:
I missed that nuance! I've revised the comment and the associated builder's guide entry to make it more explicit.

Reply:
OK, good enough, so I will resolve. (I wouldn't have used the phrase "values taken directly" myself because in some sense that phrase presumes one might take values directly, which one _should not do_. But  overall, the paragraph is clear enough in discouraging any effort to use those values directly, so not worth quibbling in the effort to get this all in to main rapidly.)